### PR TITLE
fluffychat: 1.2.0 -> 1.8.0

### PR DESCRIPTION
**Pull request by NickCao**
###### Description of changes
https://github.com/NixOS/nixpkgs/issues/163515

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by NickCao** [comment]
Looks like the vendorHash is not reproducible.

**Comment by genofire** [comment]
would you like to add yourself as an maintainer ?

**Comment by NickCao** [comment]
Got a diffoscope comparison of the outputs, the difference is within `.pub-cache/git/cache/`, where a few git repos resides.

**Comment by NickCao** [comment]
Addressed the nondeterminism.

**Comment by NickCao** [comment]
Result of `nixpkgs-review pr 210287` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>firmware-updater</li>
    <li>fluffychat</li>
    <li>yubioath-flutter</li>
  </ul>
</details>

**Review by mkg20001 — APPROVED** [review]
LGTM

**Comment by NickCao** [comment]
@ofborg build fluffychat

**Comment by NickCao** [comment]
The vendor hash mismatched once again.

**Comment by gilice** [comment]
What? Do they continue to push tags even after a release?

**Comment by NickCao** [comment]
It's about flutter, which does not natively have a `vendor` command, that we have to hack one. And it fetches a lot of git repositories, which are hard to make reproducible.

**Comment by gilice** [comment]
Still don't really get it. I understand that we have to build a `PUB_CACHE` ourselves, but what makes it this indeterministic? Aren't the git repos pinned by commit?

**Comment by NickCao** [comment]
Even if pinned by commit, the way git stores the blobs/packs and many other things can differ from run to run.

**Comment by gilice** [comment]
Weird. Do you have any ideas on fixing this?

**Comment by NickCao** [comment]
This is more of a feature report than a bug report. Also flutter does not seem to be happy with others fiddling with its caches.

**Comment by gilice** [comment]
I am sorry for bothering you so much. Still, why can't we do something like `find -type d -name '.git' -exec rm 
{} \;`? We don't need the git log of deps anyway.

**Comment by NickCao** [comment]
flutter would do a checkout itself at build time, thus `.git` has to be preserved.

**Comment by gilice** [comment]
Does this mean the build of the flutter package itself (currently green), or every flutter app?

**Comment by NickCao** [comment]
For most flutter apps.

File: pkgs/applications/networking/instant-messengers/fluffychat/default.nix [hunk 1]
```diff
@@ -1,22 +1,28 @@
...
+, fribidi
 , imagemagick
 , makeDesktopItem
 }:
 
-flutter2.mkFlutterApp rec {
+flutter.mkFlutterApp rec {
   pname = "fluffychat";
-  version = "1.2.0";
+  version = "1.8.0";
```
Comment (genofire) [inline pkgs/applications/networking/instant-messengers/fluffychat/default.nix:14]:
```suggestion
 version = "1.11.1";
```

**Comment by NickCao** [comment]
This pr is out of date has we switched to using upstream binary from https://github.com/NixOS/nixpkgs/commit/01b3e4487a565e6c39f6351099469e71603368c1. However with the recent fix in the flutter infrastructure, we may attempt an up to date source build at a later time.

**Pull request closed without merging**